cost-analyzer/templates/cost-analyzer-smtp-configmap.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if .Values.kubecostProductConfigs }}
+{{- if .Values.kubecostProductConfigs.smtp }}
+{{- if .Values.kubecostProductConfigs.smtp.enabled }}
+# If the smtp.config is not specified, the configmap will not be created
+{{- if .Values.kubecostProductConfigs.smtp.config }}
+# If the secretname is specified, the configmap will not be created
+{{- if not .Values.kubecostProductConfigs.smtp.secretname }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ default "smtp-configs" .Values.smtpConfigmapName }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{ include "cost-analyzer.commonLabels" . | nindent 4 }}
+data:
+  {{- if .Values.kubecostProductConfigs.smtp.config }}
+  config: {{  .Values.kubecostProductConfigs.smtp.config | quote }}
```

Comment:
Have you seen the value which results from this? It's not stringified JSON but stringified block scalar YAML since the input value is a block scalar. You may want to use a simple map for the values and then convert to JSON in the value or just use a YAML map for the ConfigMap portion.

Reply:
I have, and it functions as expected when decoding the string as JSON. I trust your insight, so if you feel it should change, I'll change it.